ads/plista.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {checkData} from '../src/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function plista(global, data) {
+  checkData(data, ['publickey', 'widgetname']);
+  let s, g;
+  const n = 'script', d = document;
+  d.write('<div data-display="plista_widget_' + data.widgetname + '"></div>');
+  window.PLISTA = {
+    publickey: data.publickey,
+    widgets: [{
+      name: data.widgetname,
+      pre: data.urlprefix
+    }],
+    item: data.item,
+    geo: data.geo,
+    categories: data.categories,
+    noCache: true,
+    useDocumentReady: false,
+    dataMode: 'data-display'
+  };
+  s = d.createElement(n);
+  g = d.getElementsByTagName(n)[0];
+  s.async = true;
+  s.type = 'text/javascript';
+  s.src = 'https://static' + (data.countrycode ? '-' + data.countrycode : '') + '.plista.com/async.js';
```

Comment:
For paranoia reasons, please wrap `(data.countrycode ? '-' + data.countrycode : '')` in `encodeURIComponent`